keytree/src/lib.rs

```diff
@@ -37,6 +41,41 @@ impl Xprv {
     }
 }
 
+/// Returns a intermediate child pubkey.
+pub fn intermediate_key(
+    xpub: &Xpub,
+    label: &'static std::string::String,
+    data: &String,
+) -> Option<Xpub> {
+    // question: this function returns an option because it calls a function that returns an option
+    // and I didn't know what else to do. is there something better to do?
+
+    // question: is the following kosher? https://doc.dalek.rs/merlin/struct.Transcript.html#note
```

Comment:
the note means that the ZKP protocol should be composable with others. This derivation logic is not a ZKP protocol and is not meant to be composed with the externally-provided transcript. However, it is composable with the user's choice of "derivation path", by letting the user commit arbitrary structured data to the transcript in-between Keytree committing the xpub (`pt`,`dk`) and squeezing the blinding factor `f`.